src/orion/core/worker/storage_backup.py

```diff
@@ -0,0 +1,473 @@
+            experiments = sorted(experiments, key=lambda d: d["version"])[:1]
+        logger.info(
+            f"Found experiment {experiments[0]['name']}.{experiments[0]['version']}"
+        )
+    _describe_import_progress(STEP_COLLECT_EXPERIMENTS, 1, 1, progress_callback)
+
+    preparation = _prepare_import(
+        src_storage,
+        storage,
+        resolve,
+        experiments,
+        import_benchmarks,
+        progress_callback=progress_callback,
+    )
+    _execute_import(storage, *preparation, progress_callback=progress_callback)
+
+
+def _dump(src_storage, dst_storage, name=None, version=None):
+    """Dump data from source storage to destination storage.
+
+    Parameters
+    ----------
+    src_storage: BaseStorageProtocol
+        input storage
+    dst_storage: BaseStorageProtocol
+        output storage
+    name:
+        (optional) if provided, dump only data related to experiment with this name
+    version:
+        (optional) version of experiment to dump
+    """
+    # Get collection names in a set
+    if name is None:
+        # Nothing to filter, dump everything
+        # Dump benchmarks
+        logger.info("Dumping benchmarks")
+        for benchmark in src_storage.fetch_benchmark({}):
+            dst_storage.create_benchmark(benchmark)
+        # Dump experiments
+        logger.info("Dumping experiments, algos and trials")
+        for i, src_exp in enumerate(src_storage.fetch_experiments({})):
+            logger.info(f"Dumping experiment {i + 1}")
+            _dump_experiment(src_storage, dst_storage, src_exp)
+    else:
+        # Get experiments with given name
+        query = {"name": name}
+        if version is not None:
+            query["version"] = version
+        experiments = src_storage.fetch_experiments(query)
+        if not experiments:
+            raise DatabaseError(
+                f"No experiment found with query {query}. Nothing to dump."
+            )
+        if len(experiments) > 1:
+            exp_data = sorted(experiments, key=lambda d: d["version"])[0]
+        else:
+            (exp_data,) = experiments
+        logger.info(f"Found experiment {exp_data['name']}.{exp_data['version']}")
+        # Dump selected experiments and related data
+        logger.info(f"Dumping experiment {name}")
+        _dump_experiment(src_storage, dst_storage, exp_data)
+
+
+def _dump_experiment(src_storage, dst_storage, src_exp):
+    """Dump a single experiment and related data from src to dst storage."""
+    algo_lock_info = src_storage.get_algorithm_lock_info(uid=src_exp["_id"])
+    logger.info("\tGot algo lock")
+    # Dump experiment and algo
+    dst_storage.create_experiment(
+        src_exp,
+        algo_locked=algo_lock_info.locked,
+        algo_state=algo_lock_info.state,
+        algo_heartbeat=algo_lock_info.heartbeat,
+    )
+    logger.info("\tCreated exp")
+    # Dump trials
+    for trial in src_storage.fetch_trials(uid=src_exp["_id"]):
```

Comment:
Is it possible that the experiment id changes (if `create_experiment` assigns a new one) and then the trials are pointing to the wrong experiment id? We should be able to test this easily if we dump an experiment with id > 1 as it would have id == 1 in the `dst_storage` if the id is not kept in `create_experiment`.